common/djangoapps/student/views.py

```diff
@@ -761,7 +761,15 @@ def _allow_donation(course_modes, course_id):
     donations_enabled = DonationConfiguration.current().enabled
     is_verified_mode = CourseMode.has_verified_mode(course_modes[course_id])
     has_payment_option = CourseMode.has_payment_options(course_id)
-    return donations_enabled and not is_verified_mode and not has_payment_option
+    return_val = False
```

Comment:
I think we can really just check the minimum price. Instead of lines 764 to 772, just do
return enrollment.mode.min_price == 0